arch/sim/src/sim/up_interruptcontext.c

```diff
@@ -43,10 +43,49 @@
 #include <nuttx/arch.h>
 #include "up_internal.h"
 
+#ifdef CONFIG_SIM_PREEMPTIBLE
+uint32_t host_signal_save(void);
+void host_signal_restore(uint32_t flags);
+int up_cpu_simulated_interrupt(void);
+#else
+#  define host_signal_save()           0
+#  define host_signal_restore(f)       (void)(f)
+#  define up_cpu_simulated_interrupt() 0
+#endif
+
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: up_irq_save
+ *
+ * Description:
+ *   Disable interrupts and returned the mask before disabling them.
+ *
+ ****************************************************************************/
+
+irqstate_t up_irq_save(void)
+{
+  return host_signal_save();
```

Comment:
That's correct, I'm thinking of moving the up_irq_save/up_irq_restore implementation to a separate patch as it can be decoupled from this large change

Reply:
Yes, it's better to move up_irq_save/up_irq_restore to another patch.